File: leetcode/python/0492_construct_the_rectangle.py
class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        # initialize width and length with square root of target area rounded down and up, respectively
        width = int(math.sqrt(area))
        length = int(math.ceil(area / width))
        
        # loop until width and length multiplied is equal to target area
        while width * length != area:
            # if width * length is less than target area, increment length
            if width * length < area:
                length += 1
            # if width * length is greater than target area, decrement width
            else:
                width -= 1
        
        # return [length, width]
        return [length, width]

File: leetcode/java/0492_construct_the_rectangle.java
class Solution {
    public int[] constructRectangle(int area) {
        int diff = Integer.MAX_VALUE;
        int[] res = new int[2];
        // loop through 1 to area using binary search
        // if area % middle == 0:
        //  if (diff < Math.abs((area / middle) - middle)):
        //  diff = Math.abs((area / middle) - middle)
        //  res[0] = area / middle
        //  res[1] = middle
        // return res

        for (int i = 1; i <= area; i++) {
            if (area % i == 0) {
                if (diff > Math.abs((area / i) - i)) {
                    diff = Math.abs((area / i) - i);
                    res[0] = area / i;
                    res[1] = i;
                }
            }
        }

        return res;
    }
}